app/code/Magento/Catalog/Model/Category/Toolbar/Config.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Catalog\Model\Category\Toolbar;
+
+use Magento\Catalog\Model\Category\Config as CategoryConfig;
+use Magento\Catalog\Model\CurrentCategory;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+class Config
+{
+    /**
+     * @var CurrentCategory
+     */
+    private $currentCategory;
+
+    /**
+     * @var CategoryConfig
+     */
+    private $categoryConfig;
+
+    /**
+     * Config constructor.
+     * @param CurrentCategory $currentCategory
+     * @param CategoryConfig $categoryConfig
+     */
+    public function __construct(
+        CurrentCategory $currentCategory,
+        CategoryConfig $categoryConfig
+    ) {
+        $this->currentCategory = $currentCategory;
+        $this->categoryConfig = $categoryConfig;
+    }
+
+    /**
+     * Returns a category default_sort_by attribute, or default sort field from the configuration.
+     *
+     * @return string
+     */
+    public function getOrderField(): string
```

Comment:
I was thinking how to improve this implementation to be simpler and easier to follow. Here's the version I could come up with that avoid duplication of the default behavior.

```php
        try {
            $category = $this->currentCategory->get();
            if ($category->getDefaultSortBy()) {
                return $category->getDefaultSortBy();
            }
        } catch (NoSuchEntityException $exception) {
            // Fallback to the default configuration
        }
        return $this->categoryConfig->getDefaultSortField();
```